### PR TITLE
Added hook for reading CyTRICS format input SBOMs

### PR DESCRIPTION
**Pull request by mws180000**
Implemented hook for reading input SBOMs that are in the (default) CyTRICS format.

TODO: Add additional hooks for input SBOMs in SPDX and CycloneDX formats.

### CODE REVIEW

File: surfactant/cmd/generate.py [hunk 1]
```diff
@@ -186,7 +231,7 @@ def sbom(
     if not input_sbom:
         new_sbom = SBOM()
     else:
-        new_sbom = SBOM.from_json(input_sbom.read())
+        new_sbom = input_reader.read_sbom(input_sbom)
```
Comment (nightlark) [inline surfactant/cmd/generate.py:234]: Can you make this same change for the merge command in https://github.com/LLNL/Surfactant/blob/CYT-470-add-hook-for-reading-sbom-files/surfactant/cmd/merge.py#L33?

Comment (mws180000) [reply surfactant/cmd/generate.py:234]: I have finished making those change to the merge command. For now, should we assume that all the input SBOMs being merged have the same format?

Comment (nightlark) [reply surfactant/cmd/generate.py:234]: Yes, for now let’s assume they all have the same format, and update docs or argument descriptions to mention that.

File: surfactant/cmd/merge.py [hunk 2]
```diff
@@ -20,17 +21,39 @@
...
+                if plugin.short_name().lower() == input_format.lower() and hasattr(
+                    plugin, "read_sbom"
+                ):
+                    input_reader = plugin
+                    break
+            except AttributeError:
+                pass
+
+    if input_reader is None:
+        logger.error(f'No output format "{input_format}" found')
```
Comment (nightlark) [inline surfactant/cmd/merge.py:51]:
```suggestion
        logger.error(f'No input reader plugin for format "{input_format}" found')
```

Comment (nightlark) [reply surfactant/cmd/merge.py:51]: Note: This pattern of searching for input/output readers seems like something that is going to come up in several places -- may be useful to consider moving most of the code into a plugin manager helper function that searches for a plugin with the given plugin name and implementing a given hooked function name ("read_sbom").

File: surfactant/cmd/generate.py (hunk 1 above)
Comment (mws180000) [reply surfactant/cmd/generate.py:234]: I have updated the relevant READMEs

File: surfactant/cmd/merge.py (hunk 2 above)
Comment (mws180000) [reply surfactant/cmd/merge.py]: I have moved the code for finding input readers and output writers into a separate function in manager.py

**Review by nightlark — APPROVED** [review]

**Pull request merged**